lxd/response/response.go

```diff
@@ -29,6 +29,71 @@ type Response interface {
 	String() string
 }
 
+// Devlxd response.
+type devLxdResponse struct {
+	content     any
+	code        int
+	contentType string
+}
+
+func (r *devLxdResponse) Render(w http.ResponseWriter) error {
+	var err error
+
+	if r.code != http.StatusOK {
+		http.Error(w, fmt.Sprintf("%s", r.content), r.code)
+	} else if r.contentType == "json" {
+		w.Header().Set("Content-Type", "application/json")
+
+		var debugLogger logger.Logger
+		if debug {
+			debugLogger = logger.Logger(logger.Log)
+		}
+
+		err = util.WriteJSON(w, r.content, debugLogger)
+	} else if r.contentType != "websocket" {
+		w.Header().Set("Content-Type", "application/octet-stream")
+
+		_, err = fmt.Fprint(w, r.content.(string))
+	}
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *devLxdResponse) String() string {
+	if r.code == http.StatusOK {
+		return "success"
+	}
+
+	return "failure"
+}
+
+// DevLxdErrorResponse returns an error response. If rawResponse is true, a api.ResponseRaw will be sent instead of a minimal devLxdResponse.
+func DevLxdErrorResponse(err error, rawResponse bool) Response {
+	if rawResponse {
+		return SmartError(err)
+	}
+
+	statusErr, ok := err.(api.StatusError)
```

Comment:
This is fine, but if you wanted you could also use `code, matched := api.StatusErrorMatch(err)`